app/org/sagebionetworks/bridge/stormpath/StormpathAccount.java

```diff
@@ -53,15 +56,17 @@
     private final StudyIdentifier studyIdentifier;
     private final SortedMap<Integer, BridgeEncryptor> encryptors;
     private final String healthIdKey;
-    private final String consentSignatureKey;
-    private final String consentSignaturesKey;
     private final String oldHealthIdVersionKey;
     private final String oldConsentSignatureKey;
     private final Set<Roles> roles;
+    /**
+     * All signatures saved under a key that includes the subpopulation GUID. Don't
+     * use SubpopulationGuid interface as the key, because different implementations have different
+     * equals/hashCode values. Use the GUID string directly.
+     */
+    private Map<String, List<ConsentSignature>> allSignatures;
```

Comment:
This is another reason why having SubpopGuid being separate from Subpop is a good idea. If we can guarantee that SubpopGuid is immutable, we can use it as the map key.